app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -245,7 +243,7 @@ def server_status
     def facts
       User.as_anonymous_admin do
         sync_task(::Actions::Katello::Host::Update, @host, rhsm_params)
-        @host.subscription_facet.update_facts(rhsm_params[:facts]) unless rhsm_params[:facts].blank?
+        Katello::Host::SubscriptionFacet.update_facts(@host, rhsm_params[:facts])
```

Comment:
The reason is because facts have to be imported before the host is created.  Maybe is hould move this to the RhsmFactImporter class ?   Probably makes more sense there if its not on an instance